### PR TITLE
Add documentation for the crate & repo and prepare it for publishing

### LINKED ISSUE #102: Add docs for the Blindbid library.
The library needs to be documented properly. That includes not only lib docs but also README, badges and other cosmetics of the repo.

### PR DESCRIPTION
**Pull request by CPerezz**
Cosmetics added for the repo.

Can't publish since we still have dependencies that are in git. And can't be published.
The only sad thing is that I needed to disable the doc tests since are `feature` dependant and this requires to use unstable Cargo features.

The main problem with this is basically that `-Z` can only be passed via CLI. But I wanted to add them on `config.toml` otherways it's not really practical. So for now is dissabled.

Closes #102

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -44,5 +57,4 @@ canon = [
     "canonical_derive",
     "poseidon252/canon",
     "dusk-pki/canon",
-]
-
+]
```
Comment (ZER0) [inline Cargo.toml:60]: Add a new line.

**Review by ZER0 — APPROVED** [review]
The documentation is great, just few nits. But if you prefer having a follow up issue to fix those I'm fine as well, nothing critical.

Just be careful with the new line at the end of the file.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,129 @@
...
+In order to participate in the SBA consensus, Block generators have to
+submit a bid in DUSK. As long as their bid is active - and their full-node
+is connected with the internet and running- they are participating in the
+consensus rounds. Essentially, every time a consensus round is run, the
+Block Generator software generates a comprehensive zero-knowledge proof, and
+executes various steps in order to generate a valid candidate block, and
+compete with the other Block Generators for a chance to become the winner of
+the consensus round.
+
+![](https://public.bnbstatic.com/static/research/static/images/projects/dusk-network/image18.png)
```
[the remark this answers is not in the document]
Comment (CPerezz) [reply README.md:18]: Will be addressed with #110

File: README.md [hunk 3]
```diff
@@ -1,4 +1,129 @@
...
+
+1. Blind Bid circuit;
+2. Score Generation circuit.
+
+Below we describe the Blind Bid circuit and the score generation circuit,
+who together form the pillars of the Proof-of-Blind Bid procedure.
+
+#### Blind Bid Circuit
+
+![Fig1](https://lh4.googleusercontent.com/oPR_7LLAAj6K8qjxiqMMavfucdslgv3DAGcElrn6hwDLMk0mYucfcwPyqLoT0HIyqvqmUDof4PhnXFV6-3tbImYtdG4WNHJcq72GaLsHHKE4EMiIv8pMJqLplGVlzuK6nIYsUp_r)
```
[the remark this answers is not in the document]
Comment (CPerezz) [reply README.md:54]: Will be addressed with #110

File: README.md [hunk 4]
```diff
@@ -1,4 +1,129 @@
...
+sure that the Bid is valid and eligible during the current consensus round
+and steps. Finally, in proofs 7. & 8. we check the hash of the secret (H(k))
+and the prover ID (i), asking for proof that the block generator - who we
+assume has posted the bid -, indeed is the owner of the bid.
+
+Once the process above has been completed we move to Score Generation.
+
+#### Score Generation Circuit
+
+![Fig2](https://lh5.googleusercontent.com/I6b88LUtOM5MIcbkJu3p0CZdoz34kBl7PXFovQbt4uiuAExOyW0yfLsI-1BbnzOh0u_kDv5LOB0ztuJgrh7h3y2Qh94qNN4FnP6P2Yi0cxHjJhWC3sSA49jEZtYLSutXIDkwC4Dz)
```
[the remark this answers is not in the document]
Comment (CPerezz) [reply README.md:84]: Will be addressed with #110

File: src/lib.rs [hunk 5]
```diff
@@ -4,9 +4,134 @@
...
+//! # Contributing
+//! - If you want to contribute to this repository/project please, check [CONTRIBUTING.md](https://github.com/dusk-network/dusk-blindbid/blob/master/CONTRIBUTING.md)
+//! - If you want to report a bug or request a new feature addition, please open
+//!   an issue on this repository.
+
 #![allow(non_snake_case)]
 #![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(docsrs, feature(doc_cfg))]
+#![doc(
+    html_logo_url = "https://lh3.googleusercontent.com/SmwswGxtgIANTbDrCOn5EKcRBnVdHjmYsHYxLq2HZNXWCQ9-fZyaea-bNgdX9eR0XGSqiMFi=w128-h128-e365",
```
[the remark this answers is not in the document]
Comment (CPerezz) [reply src/lib.rs:131]: Sure. I just took it from where I could.

**Pull request merged**